.bumpversion.cfg

```diff
@@ -2,11 +2,15 @@
 current_version = 0.0.31
 commit = True
 message = Update version {current_version} -> {new_version}
-parse = ^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:[\-\.]?(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$
-serialize = 
-	{major}.{minor}.{patch}{prerelease}+{buildmetadata}
-	{major}.{minor}.{patch}{prerelease}
-	{major}.{minor}.{patch}+{buildmetadata}
+parse = ^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?P<prerelease>[a|b|rc](?:0|[1-9]\d*))?(?P<devrelease>(?:\.dev)(?:0|[1-9]\d*))?(?P<buildmetadata>\+[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*)?$
```

Comment:
Leading zeroes - it can be either:
`0`
or
"`1-9` followed by any digit"
so it allows e.g. `0` and `10`, but not `01`

Strictly we _could_ allow leading zeroes in the devrelease number, but since we use a timestamp for that anyway we don't pass a number with a leading zero. 

See also:
[PEP440 Integer normalization](https://www.python.org/dev/peps/pep-0440/#integer-normalization)
[Semantic versioning regex](https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string)

Reply:
Thank you!